eip_operator/src/controller/pod.rs

```diff
@@ -95,12 +113,12 @@ impl k8s_controller::Context for Context {
         client: Client,
         pod: &Self::Resource,
     ) -> Result<Option<Action>, Self::Error> {
-        let name = pod.metadata.name.as_ref().ok_or(Error::MissingPodUid)?;
+        let name = pod.name_unchecked();
 
         let eip_api = Api::<Eip>::namespaced(client.clone(), &pod.namespace().unwrap());
 
-        let all_eips = eip_api.list(&ListParams::default()).await?.items;
-        let eip = all_eips.into_iter().find(|eip| eip.matches_pod(name));
+        let matched_eips = eip_api.list(&ListParams::default()).await?.items;
```

Comment:
Nitpick, as above, this is all eips, not just matched ones. Perhaps combine these two lines by chaining the `into_iter` to avoid the confusion?